### ISSUE TITLE
CachingDataLoader: Delete cache entry on error loading the value

### ISSUE THREAD

**Issue by mihails-strasuns** [sociomantic-tsunami/ocean#596]
@david-eckardt-sociomantic : 

Currently when <code>CachingDataLoader.load()</code> obtains an element from the external source, it stores it in the cache, then attempts to deserialise the element value. However, deserialisation may fail if the obtained value data are invalid or incomplete. <code>CachingDataLoader.loadRaw()</code> will throw an exception in that case.
If this happens, currently the element is kept in the cache but it should be deleted.

**Comment by mihails-strasuns** [comment]
This is old bug report which is no longer relevant - `CachingStructLoader` (which replaced `CachingDataLoader`) creates entry only after callback is called by `getData` implementation which will not happen if exception fires before that.
